src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.cc

```diff
@@ -492,6 +553,25 @@ XdsOverrideHostLb::SubchannelWrapper::~SubchannelWrapper() {
   }
 }
 
+void XdsOverrideHostLb::SubchannelWrapper::WatchConnectivityState(
+    std::unique_ptr<ConnectivityStateWatcherInterface> watcher) {
+  auto watcher_id = watcher.get();
+  auto wrapper =
+      std::make_unique<ConnectivityStateWatcher>(std::move(watcher), Ref());
```

Comment:
It looks like this PR is adding support for tracking the subchannel's connectivity state but doesn't yet include the EDS health state.  Are you planning to add that in this PR or do that in a separate PR?

Regardless of whether that's part of this PR or not, it's worth noting that when we do add support for endpoints in EDS health state DRAINING, we will need to watch the subchannel's connectivity state even after the child policy has stopped monitoring it.  Given that, instead of wrapping each watcher from the child policy, I think we should start our own watcher on the wrapped subchannel as soon as it is created, and when the child policy starts a watch, we can just propagate notifications from our watcher to the child policy's watcher.  We should not need to start or stop our watcher when the child policy starts or stops its watcher.